src/logger.ts

```diff
@@ -15,22 +15,31 @@ const myFormat = format.printf(info => {
 });
 
 const combinedFormats = format.combine(format.timestamp(), myFormat);
-const logger = createLogger({
-    level: "info",
-    format: combinedFormats,
-    transports: [
-        new transports.File({ dirname: logDir, filename: "error.log", level: "error" }),
-        new transports.File({ dirname: logDir, filename: "info.log", level: "info" }),
-        new transports.File({ dirname: logDir, filename: "debug.log", level: "debug" })
-    ]
-});
-// console log if we're not in production
-if (process.env.NODE_ENV !== "production" && process.env.NODE_ENV !== "test") {
-    logger.add(
-        new transports.Console({
-            format: combinedFormats
-        })
-    );
+
+export function createNamedLogger(name: string | null) {
+    const prefix = name !== null ? name + "-" : "";
+    const newLogger = createLogger({
+        level: "info",
+        format: combinedFormats,
+        transports: [
+            new transports.File({ dirname: logDir, filename: `${prefix}error.log`, level: "error" }),
+            new transports.File({ dirname: logDir, filename: `${prefix}info.log`, level: "info" }),
+            new transports.File({ dirname: logDir, filename: `${prefix}debug.log`, level: "debug" })
```

Comment:
I'm not sure why, but this doesnt appear to be working. Logs are appearing for debug - but they only contain info messages. Not sure if this was introduced by this PR, might have been a bug already. If so raise a bug ticket for it separately.

Reply:
It seems to be working as intended to me: "-error" only contains errors, "-info" contains errors and infos, while "-debug" contains everything. (see for example `watcher-debug.log` that has all the three levels).

Reply:
Yeah, that's true, I was conflating two things.

The other thing then is that we appear to logging with level: "info" - so why should the debug files be created at all?

Reply:
Not sure I get what you mean. "debug" is the most verbose level, and we currently don't have it in many places, but the watcher is using it to show detailed Event information, which is about 12 lines of output.
Or would you rather only have "error" and "info" to reduce the number of log files?

Reply:
Ah no, it's fine to have debug. I'm just surprised that the logs are showing up.

```
createLogger({
level: "info",
format: combinedFormats,
```

I've looked up the documentation for `logger: info` and apparently that's supposed to be the minimum that's being logged. So I was wondering why debug files were being created when we have the minimum set to 'info' above.

Reply:
Anyways, I'm sure this is a separate issue from your PR